lib/bolt/transport/winrm/connection.rb

```diff
@@ -355,16 +355,23 @@ def make_tempdir
         end
 
         def with_remote_file(file)
-          ext = File.extname(file)
+          if file.is_a?(Hash)
```

Comment:
Can we make this handling somewhat consistent between the SSH and WinRM implementations?